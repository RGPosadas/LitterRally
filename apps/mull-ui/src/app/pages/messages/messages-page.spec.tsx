import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MessagesPage from './messages-page';

describe('MessagesPage', () => {
  it('should render successfully', () => {
    const history = createMemoryHistory();
    const { baseElement } = render(
      <Router history={history}>
        <MessagesPage children="SwipeableRoutes" />
      </Router>
    );

    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const history = createMemoryHistory();
    const tree = renderer
      .create(
        <Router history={history}>
          <MessagesPage children="SwipeableRoutes" />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
