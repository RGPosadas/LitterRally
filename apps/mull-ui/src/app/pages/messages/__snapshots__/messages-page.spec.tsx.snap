// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesPage should match snapshot 1`] = `
<div>
  <div
    className="header undefined"
  >
    <button
      className="hamburger-button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars fa-w-14 fa-2x hamburger-menu"
        color="grey"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <h1>
      Clean up Rogers Park
    </h1>
  </div>
  <div
    className="sub-nav-bar-container with-header top-nav-bar-shadow"
  >
    <div
      className="inner-cont"
    >
      <a
        aria-current={null}
        className="subnavigation-link"
        data-testid="subnavigation-groupchat-button"
        href="/messages/group-chat"
        onClick={[Function]}
      >
        Group Chat
      </a>
      <a
        aria-current={null}
        className="subnavigation-link"
        data-testid="subnavigation-announcements-button"
        href="/messages/announcements"
        onClick={[Function]}
      >
        Announcements
      </a>
    </div>
  </div>
  SwipeableRoutes
  <form
    className="chat-input-container"
  >
    <div
      className="custom-file-upload-container"
    >
      <label
        className=""
        htmlFor="imageFile"
      >
        <div
          className=""
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-images fa-w-18 custom-file-upload-icon"
            data-icon="images"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M480 416v16c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V176c0-26.51 21.49-48 48-48h16v208c0 44.112 35.888 80 80 80h336zm96-80V80c0-26.51-21.49-48-48-48H144c-26.51 0-48 21.49-48 48v256c0 26.51 21.49 48 48 48h384c26.51 0 48-21.49 48-48zM256 128c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-96 144l55.515-55.515c4.686-4.686 12.284-4.686 16.971 0L272 256l135.515-135.515c4.686-4.686 12.284-4.686 16.971 0L512 208v112H160v-48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </label>
      <input
        accept="image/*"
        className="custom-file-upload-input"
        data-testid="file"
        id="imageFile"
        onChange={null}
        type="file"
      />
    </div>
    <div
      className="custom-text-input-container "
    >
      <label
        className="custom-text-input-label"
        htmlFor="description"
      >
        
      </label>
      <input
        className="custom-text-input input-border "
        data-testid="custom-text-input"
        id="description"
        name="description"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        className="chat-input-button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-paper-plane fa-w-16 send-icon"
          data-icon="paper-plane"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M476 3.2L12.5 270.6c-18.1 10.4-15.8 35.6 2.2 43.2L121 358.4l287.3-253.2c5.5-4.9 13.3 2.6 8.6 8.3L176 407v80.5c0 23.6 28.5 32.9 42.5 15.8L282 426l124.6 52.2c14.2 6 30.4-2.9 33-18.2l72-432C515 7.8 493.3-6.8 476 3.2z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </form>
</div>
`;
